### PR TITLE
Hide collapse icon in diff with no lines

### PR DESCRIPTION
**Pull request by JakobDev**
Sometimes (e.g. renaming a file) nothing of the content is changed. In this case, it makes no sense to show the collapse icon in a diff.

| Before        | After     |
|--------------|-----------|
|Before|After|

### CODE REVIEW

File: templates/repo/diff/box.tmpl [hunk 1]
```diff
@@ -66,13 +66,15 @@
...
 					<h4 class="diff-file-header sticky-2nd-row ui top attached normal header df ac sb">
 						<div class="df ac">
-							<a role="button" class="fold-file muted mr-2">
-								{{if $file.ShouldBeHidden}}
-									{{svg "octicon-chevron-right" 18}}
-								{{else}}
-									{{svg "octicon-chevron-down" 18}}
-								{{end}}
-							</a>
+						  {{if or (gt $file.Addition 0) (gt $file.Deletion 0)}}
```
Comment (delvh) [inline templates/repo/diff/box.tmpl]: I can see this being quite error-prone.
The first things that are already broken are non-SVG images such as pngs.
Other media and binary types will probably break just as easily, i.e. audio or video files.
I think it is better to define it as a function that takes the given file as input, and outputs a boolean.

Comment (JakobDev) [reply templates/repo/diff/box.tmpl]: Thanks for bringing that to my attention. I haven't thought about that. Fortunately, there is already a variable set to show is a file is a binary file, so this was pretty easy to solve. Everything works now as expected.

**Review by delvh — APPROVED** [review]
Yes, that'll generally work.
The only problem I can see is that for unsupported file types it'll still show `Binary file not shown` only after you want to expand it, but that's a problem for another time.

**Comment by JakobDev** [comment]
Is there still interest in this PR?

**Comment by lunny** [comment]
Yes, please resolve the conflicts.

**Review by lafriks — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Comment by wxiaoguang** [comment]
It makes the UI misaligned.

Sometimes, the header starts with a number, sometimes, it starts with an icon button.

**Comment by zeripath** [comment]
Yup I'm not a fan of the misalignment here. We should probably add a spacer in instead of just removing it.

**Comment by wxiaoguang** [comment]
The following PR:

* #22700

**Pull request merged**